src/Strategy/RPM.hs

```diff
@@ -0,0 +1,132 @@
+{-
+Yep...  Those ones right above me.  Kinda rude.  They're actually standing on my head.
+-}
+
+discover :: HasDiscover sig m => Path Abs Dir -> m ()
+discover = walk $ \dir _ files ->
+  case find (\f -> ".spec" `isSuffixOf` fileName f) files of
+    Nothing -> pure WalkContinue
+    Just specFile -> do
+      runSimpleStrategy "rpm-spec" RustGroup $ fmap (mkProjectClosure dir) (analyze specFile)
+      pure WalkSkipAll
+
+analyze :: (Has ReadFS sig m, Has (Error ReadFSErr) sig m) => Path Rel File -> m (Graphing Dependency)
+analyze specFile = do
+  specFileText <- readContentsText specFile
+  pure . buildGraph $ getSpecDeps specFileText
+
+mkProjectClosure :: Path Rel Dir ->  Graphing Dependency -> ProjectClosureBody
+mkProjectClosure dir graph = ProjectClosureBody
+  { bodyModuleDir    = dir
+  , bodyDependencies = dependencies
+  , bodyLicenses     = []
+  }
+  where
+  dependencies = ProjectDependencies
+    { dependenciesGraph    = graph
+    , dependenciesOptimal  = NotOptimal
+    , dependenciesComplete = NotComplete
+    }
+
+toDependency :: RPMDependency -> Set SpecFileLabel -> Dependency
+toDependency pkg = foldr applyLabel Dependency 
+  { dependencyType = CargoType 
+  , dependencyName = rpmDepName pkg
+  , dependencyVersion = rpmConstraint pkg
+  , dependencyLocations = []
+  , dependencyEnvironments = []
+  , dependencyTags = M.empty
+  }
+  where
+    applyLabel :: SpecFileLabel -> Dependency -> Dependency
+    applyLabel (RequiresType env) dep = dep { dependencyEnvironments = env : dependencyEnvironments dep }
+
+buildGraph :: Dependencies -> Graphing Dependency
+buildGraph Dependencies{..} = run . withLabeling toDependency $ do
+  traverse_ direct depBuildRequires
+  traverse (flip label $ RequiresType EnvDevelopment) depBuildRequires
+
+-- buildConstraint tail = RPMConstraint version <$> comparator
+buildConstraint :: Text -> Maybe VerConstraint
+buildConstraint tail = constraint
+  where
+    (comparatorStr, rawVersion) = splitOnceOn " " $ T.strip tail
+    version = T.strip rawVersion
+    constraint = case T.strip comparatorStr of
+      "<=" -> Just $ CLessOrEq version
+      "<" -> Just $ CLess version
+      ">=" -> Just $ CGreaterOrEq version
+      ">" -> Just $ CGreater version
+      "=" -> Just $ CEq version
+      "==" -> Just $ CEq version
+      _ -> Nothing
+
+splitOnceOn :: Text -> Text -> (Text, Text)
+splitOnceOn needle haystack = (head, strippedTail)
+  where
+    len = T.length needle
+    (head, tail) = T.breakOn needle haystack
+    strippedTail = T.drop len tail
+
+getTypeFromLine :: Text -> Maybe RequiresType
+getTypeFromLine line = safeReq
+  where
+    (header, value) = splitOnceOn ": " line
+    (pkgName, rawConstraint) = splitOnceOn " " $ T.strip value
+    isSafeName name = not $ "%{" `T.isInfixOf` name
+    safeReq = if isSafeName pkgName then req else Nothing
+    req = case header of
+      "BuildRequires" -> Just . BuildRequires . RPMDependency pkgName $ buildConstraint rawConstraint
```

Comment:
This seems to be another instance where we need a style guide since I'm not sure if we should be doing repeated `$`s or `.`s. I don't think thats too important right now but we should do that soon

Reply:
This is a great example of why I prefer `.` over `$`.  Neither is very obtrusive when you need it, but if you had to pick, `.` is way less noisy in a visual sense.